encoding/codecv7_types.go

```diff
@@ -0,0 +1,499 @@
+}
+
+// newDABlockV7 is a constructor function for daBlockV7 that initializes the internal fields.
+func newDABlockV7(number uint64, timestamp uint64, baseFee *big.Int, gasLimit uint64, numTransactions uint16, numL1Messages uint16) *daBlockV7 {
+	return &daBlockV7{
+		daBlockV0: daBlockV0{
+			number:          number,
+			timestamp:       timestamp,
+			baseFee:         baseFee,
+			gasLimit:        gasLimit,
+			numTransactions: numTransactions,
+			numL1Messages:   numL1Messages,
+		},
+	}
+}
+
+func newDABlockV7WithNumber(number uint64) *daBlockV7 {
+	return &daBlockV7{
+		daBlockV0: daBlockV0{
+			number: number,
+		},
+	}
+}
+
+// Encode serializes the DABlock into a slice of bytes.
+func (b *daBlockV7) Encode() []byte {
+	daBlockBytes := make([]byte, daBlockV7BlockContextEncodedLength)
+	binary.BigEndian.PutUint64(daBlockBytes[daBlockV7OffsetTimestamp:daBlockV7OffsetBaseFee], b.timestamp)
+	if b.baseFee != nil {
+		b.baseFee.FillBytes(daBlockBytes[daBlockV7OffsetBaseFee:daBlockV7OffsetGasLimit])
+	}
+	binary.BigEndian.PutUint64(daBlockBytes[daBlockV7OffsetGasLimit:daBlockV7OffsetNumTransactions], b.gasLimit)
+	binary.BigEndian.PutUint16(daBlockBytes[daBlockV7OffsetNumTransactions:daBlockV7OffsetNumL1Messages], b.numTransactions)
+	binary.BigEndian.PutUint16(daBlockBytes[daBlockV7OffsetNumL1Messages:], b.numL1Messages)
+	return daBlockBytes
+}
+
+// Decode populates the fields of a DABlock from a byte slice.
+func (b *daBlockV7) Decode(data []byte) error {
+	if len(data) != daBlockV7BlockContextEncodedLength {
+		return fmt.Errorf("block encoding is not blockContextByteSize bytes long expected %d, got %d", daBlockV7BlockContextEncodedLength, len(data))
+	}
+
+	b.timestamp = binary.BigEndian.Uint64(data[daBlockV7OffsetTimestamp:daBlockV7OffsetBaseFee])
+	b.baseFee = new(big.Int).SetBytes(data[daBlockV7OffsetBaseFee:daBlockV7OffsetGasLimit])
+	b.gasLimit = binary.BigEndian.Uint64(data[daBlockV7OffsetGasLimit:daBlockV7OffsetNumTransactions])
+	b.numTransactions = binary.BigEndian.Uint16(data[daBlockV7OffsetNumTransactions:daBlockV7OffsetNumL1Messages])
+	b.numL1Messages = binary.BigEndian.Uint16(data[daBlockV7OffsetNumL1Messages:])
+
+	return nil
+}
+
+// daChunkV7 groups consecutive DABlocks with their transactions.
+// Note: In DACodecV7 there is no notion of chunks. Blobs contain the entire batch data without any information of Chunks within.
+// However, for compatibility reasons DAChunks are still used in the codebase.
+// This way we can still uniquely identify a set of blocks and their L1 messages via their hash.
+type daChunkV7 struct {
+	daChunkV1
+}
+
+// newDAChunkV1 is a constructor for daChunkV1, initializing with blocks and transactions.
+func newDAChunkV7(blocks []DABlock, transactions [][]*types.TransactionData) *daChunkV7 {
+	return &daChunkV7{
+		daChunkV1{
+			blocks:       blocks,
+			transactions: transactions,
+		},
+	}
+}
+
+// Hash computes the hash of the DAChunk data.
+func (c *daChunkV7) Hash() (common.Hash, error) {
```

Comment:
Should we remove this impl?

Reply:
It is used from the relayer to generate the hash of a chunk which is also a unique key in the DB: https://github.com/scroll-tech/scroll/blob/bc8f9dbc83e662aa06db161b0e4dc686a6b5818d/rollup/internal/utils/utils.go#L144. I added it later to make it compatible with how the relayer works right now. But as discussed before, I think it makes sense to clean up relayer and codec once the hardfork happened.